spec/puppet-server-lib/puppet/jvm/http_client_spec.rb

```diff
@@ -21,4 +21,19 @@
       expect(request_options.get_connect_timeout_milliseconds).to equal(connect_timeout)
     end
   end
+
+  context "requests" do
+    it "throw Ruby SocketErrors instead of Java exceptions" do
+      client = Puppet::Server::HttpClient.new("localhost", 0, {})
+      [lambda { client.get('/', nil) },
+       lambda { client.post('/', nil, nil) }
+      ].each do |request_block|
+        expect(&request_block).to raise_error { |error|
+          expect(error).to be_a(SocketError)
+          expect(error).to be_a(Puppet::Server::HttpClientError)
+          expect(error.message).to eq('com.puppetlabs.http.client.HttpClientException: Error executing http request')
```

Comment:
Pretty unconventional to have so many expectations in a single `it` block.  Is there a way you can see breaking these out into separate it blocks?